service/src/main/java/com/epam/esm/exception/EntityNotFoundException.java

```diff
@@ -0,0 +1,14 @@
+package com.epam.esm.exception;
+
+public class EntityNotFoundException extends RuntimeException {
+
+  private final long entityId;
```

Comment:
why do you use Class<?> causeEntity in InvalidEntityException, but here don't?

Reply:
During gift certificate creation or update user can mess up either in certificate name or in tag name. In InvalidEntityException Class<?> causeEntity helps to specify where the mistake occurred. When user searches for entity EntityNotFoundException can only be thrown for an exact entity that user was searching for. So there is no need to specify class